lib/suppressions.cpp

```diff
@@ -304,28 +304,33 @@ bool Suppressions::Suppression::parseComment(std::string comment, std::string *e
     if (comment.compare(comment.size() - 2, 2, "*/") == 0)
         comment.erase(comment.size() - 2, 2);
 
-    const std::string cppchecksuppress = "cppcheck-suppress";
+    const std::set<std::string> cppchecksuppress{
+        "cppcheck-suppress",
+        "cppcheck-suppress-begin",
+        "cppcheck-suppress-end",
+        "cppcheck-suppress-file"
+    };
 
     std::istringstream iss(comment.substr(2));
     std::string word;
     iss >> word;
-    if (word.substr(0, cppchecksuppress.size()) != cppchecksuppress)
+    if (!cppchecksuppress.count(word))
```

Comment:
That changes the behavior and will most likely impact the performance negatively. It should be using `startsWith()`.

Also if you do not use the actual contents of the `count()` result you should use `std::find()` as that stops at the first occurrence instead of continuing to look for all of them.

Reply:
> Also if you do not use the actual contents of the count() result you should use std::find() as that stops at the first occurrence instead of continuing to look for all of them.

for a std::set, it is technically the same. here is the implementation for count on my computer (/usr/include/c++/11/bits/stl_set.h):
```
           size_type
           count(const key_type& __x) const
           { return _M_t.find(__x) == _M_t.end() ? 0 : 1; }
```
and the find() also invokes the same `_M_t.find` call:
```
      const_iterator
      find(const key_type& __x) const
      { return _M_t.find(__x); }
```

Reply:
> That changes the behavior and will most likely impact the performance negatively. It should be using startsWith().

but the point is that I don't want to match random strings.. I want it to be explicit. It used to be explicit before JohanBertrand changed it..

Reply:
> for a std::set, it is technically the same.

That is true. I did not check the container.